tools/base/envoy_python.bzl

```diff
@@ -244,3 +244,90 @@ def envoy_genjson(name, srcs = [], yaml_srcs = [], filter = None, args = None):
         args = args,
         filter = filter,
     )
+
+def envoy_py_data(name, src, format = None, entry_point = base_entry_point):
+    """Preload JSON/YAML data as a python lib.
+
+    Data is loaded to python and then dumped to a pickle file.
+
+    A python lib is provided which exposes the pickled data.
```

Comment:
This is very cool. Do we want to pickle into an in-memory proto object as well, e.g. factor out things like parsing YAML into JSON into protobuf and validating?

Reply:
not sure if pickling a proto would work - json works because it can be marshalled as python primitives - im guessing that is not the case with a proto object

in the case where we currently parse yaml -> json -> protobuf -> validate -> json - ie the protodoc manifest - i have created a pipeline such that the resulting json is made available via this macro

Reply:
not entirely sure if i understood the suggestion here

re pickling - anything can theoretically be pickled - eg the ZODB is an object db entirely built on pickles (iirc), but you have to then provide marshals to load/dump types

re protobuf - i see that you can de/serialize descriptors, not sure but there might be a way the python objects could be pickled - from my reading of how protobuf works the python libs load from the descriptors dynamically - so _if_ you could pickle protobuf objects there may be some speed up by preloading this part. At the cost of considerable complexity.